### DISCUSSION TITLE
Invalid arguments specified in evaluating the interpolant

### DISCUSSION THREAD

**Discussion by v-upadhyay** [casadi/casadi#3488] [Q&A]
I have a data set of input and output, and I used the function scatter interpolation to make it continuous. When I was forming a mathematical model for a system in Casadi using casadi variable (SX.sys) , then I got an error from the interpolation function "Invalid arguments specified in evaluating the interpolant." and I was inputting the casadi variables in the interpolation function .

function out=mapping(i1,i2,data)

xd=data(:,1); 
yd=data(:,3); 
zd=data(:,2); 

F = scatteredInterpolant(xd,yd,zd);
xq = i1;                 
yq = i2;                
zq = F(xq,yq);           

m=zq;

end

This is the interpolation function. It's working when i1 and i2 are Matlab variables but not when they are casadi variables.

Thanks in advance! 

( For Reference:

I am trying to make a mathematical model of a system to apply mpc to it:
The issue is that some part of the model is mathematical so I used casadi SX variables but the other part is data-based where I have to use some interpolation function. However, I am not able to do that because the interpolation function does not support casadi SX variables as input even the inbuilt casadi interpolation function. )

**Comment by jgillis** [comment]
`scatteredInterpolant` will be one many builtin matlab functions that is only implemented for numerical types.

Your options:
 * using matlab commands, remesh yoru data onto a regular grid, and use casadi `interpolant` (see '2D lookup tables' at docs.casadi.org)
 * use a callback to inject a piece of numerical matlab code into a casadi graph (see 'Subclassing Callback' at docs.casadi.org)

Note that these strategies work with MX only.
